### PR TITLE
feat: showSizeChanger accept SelectProps

### PR DESCRIPTION
**Pull request by prakashks20**
Ant Design Pagination enhanced feature added in https://github.com/ant-design/ant-design/pull/46757 depends on this PR changes

## Summary by CodeRabbit

- **新功能**
	- 更新了分页组件的API文档，增强了可读性和一致性，特别是对`showSizeChanger`参数的详细描述。
	- 新增了`showSizeChanger`功能的演示文档和示例组件，展示了如何在用户界面中实现可调整页面大小的分页。
	- `Options`组件现在集成了`showSizeChanger`功能，支持条件显示搜索输入和通知外部组件大小变化。
- **测试**
	- 添加了针对具有大小变化功能的`Pagination`组件的单元测试，确保其正常工作。

### CODE REVIEW

**Comment by prakashmallow** [comment]
Waiting for this pull request code review. This pull request change depends on fixing the antd pagination issue.

File: docs/examples/pageSizeChanger.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      <Pagination
+        defaultCurrent={1}
+        total={50}
+        selectComponentClass={Select}
+        showSizeChanger
+        pageSizeChanger={{
+          options: [10, 25, 50, 75, 100],
+          showSearch: false,
+          onChange: pageSizeOnChange,
+        }}
```
Comment (afc163) [inline docs/examples/pageSizeChanger.tsx:22]:
```suggestion
        showSizeChanger={{
          options: [10, 25, 50, 75, 100],
          showSearch: false,
          onChange: pageSizeOnChange,
        }}
```

We could reuse the `showSizeChanger` prop for accepting object value too.

Comment (prakashmallow) [reply docs/examples/pageSizeChanger.tsx]: @afc163 @MadCcc  PR review changes updated. Could you please review these changes?

Comment (prakashmallow) [reply docs/examples/pageSizeChanger.tsx]: @MadCcc @afc163 The test case coverage failed issue has been fixed and updated on this PR. Could you please review this PR?

File: src/interface.ts [hunk 2]
```diff
@@ -32,12 +38,12 @@ export interface PaginationData {
 
   hideOnSinglePage: boolean;
   align: 'start' | 'center' | 'end';
-  showSizeChanger: boolean;
+  showSizeChanger: boolean | showSizeChangerLocale;
```
Comment (afc163) [inline src/interface.ts:41]: showSizeChanger is not locale.

Comment (afc163) [inline src/interface.ts:41]:
```suggestion
  showSizeChanger: ShowSizeChanger;
```

File: src/interface.ts [hunk 3]
```diff
@@ -17,6 +17,12 @@ export interface PaginationLocale {
   page_size?: string;
 }
 
+export interface showSizeChangerLocale {
+  options?: string[] | number[];
+  showSearch?: boolean;
+  onChange?: (size: number) => void;
+}
```
Comment (afc163) [inline src/interface.ts:24]:
```suggestion
type showSizeChangerLocale = boolean | {
  options?: string[] | number[];
  showSearch?: boolean;
  onChange?: (size: number) => void;
}
```

File: src/interface.ts (hunk 2 above)
Comment (prakashmallow) [reply src/interface.ts:41]: Will change it as per your suggestion

File: src/interface.ts (hunk 3 above)
Comment (prakashmallow) [reply src/interface.ts:24]: Will change it as per your suggestion

**Comment by prakashmallow** [comment]
@afc163 Code review changes have been addressed. Please review these changes.

**Comment by prakashmallow** [comment]
@afc163 @MadCcc @zombieJ Waiting for this change to fix the antd pagination size selector issue. Could you please review this pull request changes

**Comment by prakashks20** [comment]
@afc163 @MadCcc @zombieJ  Waiting for the changes in this pull request to fix the Antd pagination size selector issue. Could you please review it as soon as possible?

File: src/Pagination.tsx [hunk 4]
```diff
@@ -588,9 +588,11 @@ const Pagination: React.FC<PaginationProps> = (props) => {
         selectPrefixCls={selectPrefixCls}
         changeSize={showSizeChanger ? changePageSize : null}
         pageSize={pageSize}
-        pageSizeOptions={pageSizeOptions}
+        pageSizeOptions={(typeof showSizeChanger === 'object' && showSizeChanger?.options) || pageSizeOptions}
         quickGo={shouldDisplayQuickJumper ? handleChange : null}
         goButton={gotoButton}
+        onChange={typeof showSizeChanger === 'object' && showSizeChanger?.onChange}
```
Comment (afc163) [inline src/Pagination.tsx:594]: changeSize and onChange is basicly same function, we should unify them to one

File: src/Pagination.tsx [hunk 5]
```diff
@@ -588,9 +588,11 @@ const Pagination: React.FC<PaginationProps> = (props) => {
         selectPrefixCls={selectPrefixCls}
         changeSize={showSizeChanger ? changePageSize : null}
         pageSize={pageSize}
-        pageSizeOptions={pageSizeOptions}
+        pageSizeOptions={(typeof showSizeChanger === 'object' && showSizeChanger?.options) || pageSizeOptions}
```
Comment (afc163) [inline src/Pagination.tsx:591]: `showSizeChanger` should be assigned to a default object value if `showSizeChanger={true}`. Then we don't need `typeof showSizeChanger` here.

File: src/Options.tsx [hunk 6]
```diff
@@ -59,6 +63,9 @@ const Options: React.FC<OptionsProps> = (props) => {
 
   const changeSizeHandle = (value: number) => {
     changeSize?.(Number(value));
+    if (onChange && typeof onChange === 'function') {
+      onChange?.(Number(value));
+    }
```
Comment (afc163) [inline src/Options.tsx:68]: `changeSize` and `onChange` is basicly same function, we should unify them to one.

File: src/Pagination.tsx (hunk 4 above)
Comment (prakashks20) [reply src/Pagination.tsx:594]: @afc163 Can I remove onChange from showSizeChanger?

Comment (afc163) [reply src/Pagination.tsx:594]: LGTM, Pagination's `onChange` could be enough in most cases.

File: src/Options.tsx [hunk 7]
```diff
@@ -128,7 +135,7 @@ const Options: React.FC<OptionsProps> = (props) => {
       <Select
         disabled={disabled}
         prefixCls={selectPrefixCls}
-        showSearch={false}
+        showSearch={showSearch}
         className={`${prefixCls}-size-changer`}
         optionLabelProp="children"
         popupMatchSelectWidth={false}
```
Comment (afc163) [inline src/Options.tsx:141]: Could we pass the whole showSizeChanger object to Select? That we could customize Select as needed.

```jsx
showSizeChanger={{
  disabled: true,
}}
```

File: src/Pagination.tsx (hunk 5 above)
Comment (afc163) [reply src/Pagination.tsx:591]: ping

**Pull request merged**